examples/trainercontroller_configs/epoch-level-eval-loss-patience.yaml

```diff
@@ -0,0 +1,19 @@
+controller-metrics:
+  - name: trainer_state
+    class: TrainingState
+  - name: eval_loss_window
+    class: HistoryBasedMetric
+    arguments:
+      window-size: 1
+operations:
+  - name: hfwithpatience
+    class: HFControls
+    arguments:
+      patience-threshold: 3
+controllers:
+  - name: epoch-level-eval-loss-patience
+    triggers:
+      - on_epoch_end
+    rule: len(eval_loss_window["metrics"]) > 0 and eval_loss_window["metrics"][-1]["eval_loss"] > 2.2 and trainer_state["epoch"] > 2
```

Comment:
isn't patience an attribute at this level instead of at the control level? This allow us to not having to keep the logic of checking if a rule succeeded before invoking a control.

Reply:
Addressed.